src/transformers/data/data_collator.py

```diff
@@ -49,20 +57,23 @@ def collate_batch(self, features: List[InputDataClass]) -> Dict[str, torch.Tenso
         # on the whole batch.
         first = features[0]
 
+        # Cache to avoid class lookup
+        device = self.device
```

Comment:
is this optimization really worth it?

Reply:
.to(device) is already being done in train() method in trainer.

FYI